README.md

```diff
@@ -1,61 +1,2354 @@
+directives in the Dockerfile.
+
+Because images are formed of a set of file system layers, an image can actually
+inherit another image as its base, which therefore gives it immediate access to
+the filesystem layout that exists in the base image.
+
+Here's a small example. Consider an image, let's call it 'simple-base', which
+has the following directories and files in its layers:
+```
+.
+├── directory-1
+│   ├── directory-1-1
+│   │   └── file2.txt
+│   └── file1.txt
+└── directory-2
+    └── file3.txt
+```
+We can use the `FROM` directive at the start of a Dockerfile to inherit from
+this base image:
+```
+# Will use simple-base as a base images, so all the files/directories from
+# simple-base are now available to work on.
+FROM simple-base
+```
+(You'll note that comments can be used in Dockerfiles, by simply using `#` at
+the start of the line, very much like Linux/Unix shell operations.)
+
+Further directives in the file can carry out operations on the inherited
+filesystem, as well as creating new layers which include new or modified
+directories and files. Every time the filesystem is modified in any way in the
+Dockerfile, a new layer is created. This usually occurs after each full
+directive.
+
+As well as inheriting from base images, it's also possible to start a completely
+new Docker image which does not include any layers. This can be achieved by
+using the special image name `scratch`. This tells Docker it should try to
+inherit from any other image, but start a brand new set of layers. In fact, this
+is how a large number of 'standard' base images start, such as the Docker
+versions of Linux distributions work, by starting from `scratch` and copying in
+all of the files that build up that version of Linux.
+
+#### 4.2 [`COPY`](https://docs.docker.com/engine/reference/builder/#copy) Directive
+
+So now we now how to use a base image to start with a filesystem in our own
+image, but how do we get new files into it?
+
+This is where the `COPY` directive comes in. This allows the copying of files
+from the host's local filesystem into the Docker image being built. It's fairly
+similar to the standard copying commands under most OS variants these days in
+that it takes the location of a source file or directory on the host and a
+destination where the contents of the source should be copied inside the image.
+`COPY` preserves all flags on the file when copied, so for example if you copy
+a shell script that has the execution flag set, this will be preserved when
+copied into the image.
+
+**NOTE:** The directory location of the Dockerfile being used to build the image
+    the image is known as the build context. Any files in the build context, or
+    any of it's sub-directory hierarchy is also part of the build context.
+    This can be changed by other commands, but this is the default and we'll be
+    using it for the moment.
+
+The following will recursively copy the contents of the  `directory-3` directory
```

Comment:
Not sure about this sentence, maybe just needs some punctuation? 

The following will recursively copy the contents of the  `directory-3` directory, the current build context, into the image at the location `/directory-3`?

Reply:
I'd missed an `in`